coral/src/domain/topic/topic-transformer.ts

```diff
@@ -167,8 +168,33 @@ function transformGetTopicRequestsResponse(
   };
 }
 
+function transformTopicOverviewResponse(
```

Comment:
One thing I would love this transformer to do is make `topicPromotiomnDetails` easier to work with. Because as it stands, we need to repeatedly check if something is undefined before we can use the data we want:

- here: https://github.com/aiven/klaw/pull/1353/files#diff-06a0fe824cf3ce391d787f11417a06e0fd07662b99becbb7821bbe2366a23896R123-R126

- or here: https://github.com/aiven/klaw/pull/1353/files#diff-06a0fe824cf3ce391d787f11417a06e0fd07662b99becbb7821bbe2366a23896R68-R74

It's a little tricky, because the structure of `topicPromotiomnDetails` is like

```
topicPromotionDetails: {
  [string that is environment name, like DEV]: {
    ...data
    }
}
```

So, `topicPromotionDetails[environment]` will always be potentially undefined, as env names cannot be a strict type, I think.

Reply:
I'll look into that!

Reply:
Actually, this might not be necessary, after talking with @aindriu-aiven : he opened this issue which should simplify a lot https://github.com/aiven/klaw/issues/1361

Reply:
I left it alone for now and we can update that in a follow up if we still need to!